### PR TITLE
Fixed info banner positioning issues

### PR DESCRIPTION
**Pull request by alesker**
Refactored constraint usage and animation implementation details

### CODE REVIEW

File: Pod/Classes/AFMInfoBanner.m [hunk 1]
```diff
@@ -228,6 +249,18 @@ - (void)layoutSubviews
     [super layoutSubviews];
 }
 
+- (void)updateUI
+{
+    // First pass calculates the height correctly with existing constraints.
+    // Self-only doesn't calculate height on iOS 6, so pass through a superview
```
Comment (karpelcevs) [inline Pod/Classes/AFMInfoBanner.m]: Do we actually still support iOS 6 in there? Shouldn't we drop it in future updates?

File: Pod/Classes/AFMInfoBanner.m [hunk 2]
```diff
@@ -247,6 +280,14 @@ - (void)show
 
 - (void)show:(BOOL)animated
 {
+
+    if (self.hideTimer) {
+        [self.hideTimer invalidate];
+        self.hideTimer = nil;
+    }
+
+    if ([self superview]) return;
```
Comment (karpelcevs) [inline Pod/Classes/AFMInfoBanner.m]: Ekhem

**Comment by karpelcevs** [comment]
And long-long living bug finally gone. Awesome!

File: Pod/Classes/AFMInfoBanner.m (hunk 2 above)
Comment (alesker) [inline Pod/Classes/AFMInfoBanner.m]: Fixed

File: Pod/Classes/AFMInfoBanner.m (hunk 1 above)
Comment (alesker) [inline Pod/Classes/AFMInfoBanner.m]: Changed to `[self layoutIfNeeded]`

**Pull request merged**